routers/repo/issue.go

```diff
@@ -940,6 +956,10 @@ func UpdateIssueStatus(ctx *context.Context) {
 	}
 	for _, issue := range issues {
 		if err := issue.ChangeStatus(ctx.User, issue.Repo, isClosed); err != nil {
+			if models.IsErrDependenciesLeft(err) {
+				ctx.Error(http.StatusPreconditionFailed, "", fmt.Sprintf("cannot close this issue because it still has open dependencies"))
```

Comment:
I can't be bothered to check this issue out, but for me it seems that the response in all the cases where we have `ctx.Error(http.StatusPreconditionFailed, "",` the server gives an empty response, when I think it should give the error message. Can you check out why this happens? To reproduce, try selecting an issue (with the checkbox) with unsolved deps from the issue list on a repo and clicking the close button. The request should then be in your browser's network tab.

Reply:
Thats probably because the JS function doen't check for this. I'll dig a bit and keep you updated.

Reply:
Whenever this occures on the API, I think we're good whith this (because this means it is a client side issue). However, in the case you described, it seems to be serverside.

Reply:
In our case here, the error lies in `public/js/index.js:106`, where the function used to make the post request is declared, it doesn't check for _any_ errors... 

As fixing this would probably be time for another pr, I'd propose to just fix my code on the serverside so it returns JSON instead of plain HTML (as the  function returns JSON in the event of success) when it has dependencies left, and then open an issue + seperate pr to fix this on the clientside.

Reply:
I actually didn't mean the JS checking. Your browser should have a "Network tab" which, when devtools are enabled, will record all requests made (also by JS), both failing and succeeding, but I don't see anything there in the "Response" tab, so I figure something must be wrong when making the response on the server side?

Reply:
I've seen an HTML page there :thinking: 
But anyway, the JS-function used to make the request doesn't check for errors and therefore fails silently if the request retuns anything other than status `200` (which is IMHO the real issue here). As I said, I've changed the function on the serverside to return JSON which could then be interpreted by the JS.